src/score.cpp

```diff
@@ -0,0 +1,48 @@
+/*
+  Stockfish, a UCI chess playing engine derived from Glaurung 2.1
+  Copyright (C) 2004-2024 The Stockfish developers (see AUTHORS file)
+
+  Stockfish is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  Stockfish is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "score.h"
+
+#include <cassert>
+#include <cmath>
+#include <cstdlib>
+
+#include "uci.h"
+
+namespace Stockfish {
+
+Score::Score(Value v, const Position& pos) {
+    assert(-VALUE_INFINITE < v && v < VALUE_INFINITE);
+
+    if (std::abs(v) < VALUE_TB_WIN_IN_MAX_PLY)
+    {
+        score = InternalUnits{UCIEngine::to_cp(v, pos)};
```

Comment:
Well I'm not sure, I think the to_cp and the wdl shouldn't be a part of the uciengine altogether. If we were to make it a library then the uciengine would be something which the user implements and eval normalization and wdl isn't something they should be doing. 
Meaning I think this should be moved to another (new) namespace but still return the normalized value not the internal, or maybe the internal and normalized as a pair?

Reply:
Right, I see how this makes sense. It's a bit of a mess because conversion from internal units to cp is no longer straightforward. Not sure how the precision loss would affect people, but it should be minimal. Worst case we could convert to millipawns or smth in the future. So I guess just the naming needs fixing.